### PR TITLE
[GEOT-7265] Do not use extlink user for index page

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOT-7265](https://osgeo-org.atlassian.net/browse/GEOT-7265) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

The use of extlink user requires hand editing of the index page for prior branches.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [ ] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: docs/index/conf.py [hunk 1]
```diff
@@ -4,3 +4,8 @@
 
 html_theme='geotools-index'
 html_title='GeoTools Documentation'
+
+# do not use user extlink for index page generation
+del extlinks['user']
+
+print(extlinks)
```
Comment (mprins) [inline docs/index/conf.py:11]: this seems a debugging statement?

**Review by mprins — APPROVED** [review]

File: docs/index/index.rst [hunk 2]
```diff
@@ -23,17 +23,17 @@ User Guide
...
-:user:`User Guide <>` ( `stable <https://docs.geotools.org/stable/userguide/>`__ | `maintenance <https://docs.geotools.org/maintenance/userguide/>`__ )
+`User Guide <https://docs.geotools.org/stable/userguide/>`__ ( `stable <https://docs.geotools.org/stable/userguide/>`__ | `maintenance <https://docs.geotools.org/maintenance/userguide/>`__ )
     Tutorials, module reference and examples illustrating the uses of GeoTools. Updated with
     build instructions and advanced tutorials and integration instructions.
     
     The user guide is available for `download <https://sourceforge.net/projects/geotools/files/>`__
     as a zip file.
        
-:user:`Tutorials <tutorial/>` ( `stable <https://docs.geotools.org/stable/userguide/tutorial/>`__ | `maintenance <https://docs.geotools.org/maintenance/userguide/tutorial/>`__ )
+`Tutorials <tutorial/https://docs.geotools.org/stable/userguide/>`__ ( `stable <https://docs.geotools.org/stable/userguide/tutorial/>`__ | `maintenance <https://docs.geotools.org/maintenance/userguide/tutorial/>`__ )
```
Comment (aaime) [inline docs/index/index.rst:33]: ``<tutorial/https://docs.geotools.org/stable/userguide/>`__``

This is looking strange too...

File: docs/index/conf.py (hunk 1 above)
Comment (jodygarnett) [inline docs/index/conf.py:11]:
```suggestion
```

File: docs/index/index.rst (hunk 2 above)
Comment (jodygarnett) [inline docs/index/index.rst:33]:
```suggestion
`Tutorials <https://docs.geotools.org/latest/tutorial/https://docs.geotools.org/stable/userguide/>`__ ( `stable <https://docs.geotools.org/stable/userguide/tutorial/>`__ | `maintenance <https://docs.geotools.org/maintenance/userguide/tutorial/>`__ )
```

**Comment by jodygarnett** [comment]
Should be fixed now

**Comment by jodygarnett** [comment]
I don't know about the downstream check failing

**Comment by aaime** [comment]
Nothing should be stopping you from looking at the build logs though.

```
2022-11-17T16:47:21.0141913Z [ERROR] Tests run: 13, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 83.641 s <<< FAILURE! - in org.geowebcache.service.wmts.WMTSServiceTest
2022-11-17T16:47:21.0143107Z [ERROR] testGetCapEmptyStyleFilter(org.geowebcache.service.wmts.WMTSServiceTest)  Time elapsed: 79.038 s  <<< FAILURE!
2022-11-17T16:47:21.0144463Z junit.framework.AssertionFailedError: 
2022-11-17T16:47:21.0145403Z At line 27, column: 111 ==> schema_reference.4: Failed to read schema document 'http://www.w3.org/2001/xml.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
2022-11-17T16:47:21.0146492Z  At line 21, column: 108 ==> schema_reference.4: Failed to read schema document 'http://www.w3.org/2001/xml.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
2022-11-17T16:47:21.0147734Z  At line 19, column: 108 ==> schema_reference.4: Failed to read schema document 'http://www.w3.org/2001/xml.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
2022-11-17T16:47:21.0148719Z  At line 35, column: 108 ==> schema_reference.4: Failed to read schema document 'http://www.w3.org/2001/xml.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
2022-11-17T16:47:21.0149459Z  At line 39, column: 45 ==> src-resolve: Cannot resolve the name 'xml:lang' to a(n) 'attribute declaration' component.
2022-11-17T16:47:21.0149834Z  
2022-11-17T16:47:21.0150306Z 	at org.geowebcache.service.wmts.WMTSServiceTest.testGetCapEmptyStyleFilter(WMTSServiceTest.java:992)
```

Looks like it's a network glitch, retrieving the infamous xml.xsd. I've restarted the build.

**Pull request merged**